roles/edpm_kernel/tasks/ksm.yml

```diff
@@ -0,0 +1,61 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Enable/disable ksm
+  block:
+    - name: Disable KSM
+      when: not edpm_kernel_enable_ksm|bool
+      block:
+        - name: Check for ksm
+          ansible.builtin.shell: systemctl is-active ksm.service || systemctl is-enabled ksm.service
```

Comment:
Since you're going to stop and disable the services in the next task, and there isn't a `when` condition in this one, i'd remove this task.

Shell is one of the slowest module in ansible so in the worst case you're going to execute 2 tasks for an operation that you are doing anyway.

Reply:
we used to need this check in tripleo to trigger the package removal after disabling the service. Since we couldn't use handlers in tripleo by the design concerns, I guess. But here it seems I can remove that and notify a handler to remove the package whenever the service has been stopped